src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2674,6 +2666,106 @@ theorem getLsbD_intMax (w : Nat) : (intMax w).getLsbD i = decide (i + 1 < w) :=
 
 
+/-! ### abs -/
+
+private theorem two_pow_plus_one_div_two (w : Nat) : ((2^w + 1) / 2) = 2^(w - 1) := by
+  apply Nat.div_eq_of_lt_le
+  · rcases w with rfl | w
+    · decide
+    · simp
+      omega
+  · rcases w with rfl | w
+    · decide
+    · rw [Nat.add_one_sub_one, Nat.add_mul,
+        Nat.one_mul, Nat.add_lt_add_iff_right,
+        Nat.pow_add]
+      omega
+
+theorem abs_def {x : BitVec w} : x.abs = if x.msb then .neg x else x := rfl
+
+/-- The value of the bitvector (interpreted as an integer) is always less than 2^w -/
+theorem toInt_lt (x : BitVec w) : x.toInt < 2 ^ w := by
+  rw [toInt_eq_msb_cond]
+  norm_cast
+  omega
+
+/-- The negation value of the bitvector (interpreted as an integer) is always less than 2^w -/
+theorem neg_toInt_lt (x : BitVec w) : - x.toInt < 2 ^ w := by
+  have := toInt_lt x
+  rw [toInt_eq_msb_cond]
+  split 
+  case isTrue h => 
+    simp only [gt_iff_lt]
+    norm_cast
+    have := msb_eq_true_iff_two_mul_ge.mp h
+    omega
+  case isFalse h => 
+    norm_cast
+    omega
+
+/-- The msb of `intMin w` is `true` for all  `w > 0` -/
+@[simp]
+theorem msb_intMin : (intMin w).msb = decide (w > 0) := by
+  rw [intMin]
+  rw [msb_eq_decide]
+  simp
+  rcases w with rfl | w 
+  · rfl
+  · simp
+    have : 0 < 2^w := Nat.pow_pos (by decide)
+    have : 2^w < 2^(w + 1) := by
+      rw [Nat.pow_succ]
+      omega
+    rw [Nat.mod_eq_of_lt (by omega)]
+    simp
+
+@[simp]
+theorem abs_intMin : (intMin w).abs = intMin w := by
+  rw [abs_def]
+  simp [msb_intMin]
+
+
+@[simp] theorem toInt_zero (w : Nat) : (0#w).toInt = 0 := by
+  simp [BitVec.toInt]
+  omega
+
+/-- the msb is true iff the bitvector , when interpreted as a signed 2s complement number, is less than zero -/
+theorem msb_eq_decide_slt_zero (x : BitVec w) : x.msb = decide (x.slt 0#w) := by
+  simp only [BitVec.slt, toInt_eq_msb_cond, msb_zero, Bool.false_eq_true,
+    ↓reduceIte, toNat_ofNat, Nat.zero_mod, Int.Nat.cast_ofNat_Int, decide_eq_true_eq]
+  rcases h : x.msb  <;> simp [h] <;> omega
+
+
+theorem sle_eq_not_slt (x y : BitVec w) : x.sle y = !y.slt x := by
+  simp only [BitVec.sle, BitVec.slt, ← decide_not, decide_eq_decide]; omega
+
+
+/-- If x >= 0, then x.abs = x -/
+theorem abs_of_sle (x : BitVec w) (hx : (0#w).sle x) : x.abs = x := by
+  rw [abs_def, msb_eq_decide_slt_zero]
+  simp only [sle_eq_not_slt, Bool.not_eq_eq_eq_not, Bool.not_true] at hx
+  simp [hx]
+
+/-- If x < 0, then x.abs = -x -/
+theorem abs_of_slt (x : BitVec w) (hx : x.slt 0#w) : x.abs = -x := by
+  rw [abs_def, msb_eq_decide_slt_zero]
+  simp only [hx, decide_True, ↓reduceIte, neg_eq]
+
+/-- Either x < y or x ≥ y, i.e. y ≤ x -/
+theorem slt_or_sle (x y : BitVec w) : x.slt y ∨ y.sle x := by
+ rw [BitVec.slt, BitVec.sle]
+ by_cases h : x.toInt < y.toInt <;> simp [h] <;> omega
+
+/-- TODO: what should I name this lemmas? -/
+theorem abs_cases (x : BitVec w) : x.abs = 
+    if x = intMin w then (intMin w)
```

Comment:
These are often named `abs_eq_if`.

Also, you may not need the `x = intMin w` case, as `x = -x` holds for `intMin` as well, and so falls in your second case.